jaxrs/src/test/java/org/jboss/aerogear/unifiedpush/rest/kafka/KafkaTest.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.rest.kafka;
+
+import io.debezium.kafka.KafkaCluster;
+import io.debezium.util.Testing;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class KafkaTest {
+    private KafkaConsumer consumer;
+    private KafkaProducer producer;
+    private File dataDir;
+    private KafkaCluster cluster;
+    final private String topic = "testTopic";
+
+    @Before
+    public void beforeEach() throws IOException {
+        dataDir = Testing.Files.createTestingDirectory("cluster");
+        cluster = new KafkaCluster().usingDirectory(dataDir).withPorts(5000, 5001);
+
+        cluster.addBrokers(1).startup();
+        cluster.createTopic(topic, 1, 1);
+    }
+
+    @After
+    public void afterEach() {
+        cluster.shutdown();
+        Testing.Files.delete(dataDir);
+    }
+
+    @Test
+    public void testSendAndReceive() {
+        producer = new KafkaProducer<>(producerProps());
+        consumer = new KafkaConsumer<>(consumerProps());
+
+        for(int i = 0; i < 10; i++) {
+            producer.send(new ProducerRecord(topic, Integer.toString(i), "Message"));
+        }
+
+        consumer.subscribe(Arrays.asList(topic));
+        ConsumerRecords<String, String> records = consumer.poll(200);
+
+        for(ConsumerRecord<String, String> record : records) {
+            assertEquals(record.value(), "Message");
+        }
+    }
+
+    private Properties producerProps() {
+        Properties props = cluster.useTo().getProducerProperties(null);
+        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        return props;
+    }
+
+    private Properties consumerProps() {
+        Properties props = cluster.useTo().getConsumerProperties("10", null, OffsetResetStrategy.EARLIEST);
+        props.put("key.deserializer","org.apache.kafka.common.serialization.StringDeserializer");
+        props.put("value.deserializer","org.apache.kafka.common.serialization.StringDeserializer");
+        return props;
+    }
+}
```

Comment:
I'm going to add one or two more "check not null" tests, should I squash all the commits?